pkg/dcgm/health.go

```diff
@@ -23,14 +40,84 @@ type DeviceHealth struct {
 	Watches []SystemWatch
 }
 
-func setHealthWatches(groupId GroupHandle) (err error) {
-	result := C.dcgmHealthSet(handle.handle, groupId.handle, C.DCGM_HEALTH_WATCH_ALL)
+// HealthSet enable the DCGM health check system for the given systems
+func HealthSet(groupId GroupHandle, systems HealthSystem) (err error) {
+	result := C.dcgmHealthSet(handle.handle, groupId.handle, C.dcgmHealthSystems_t(systems))
 	if err = errorString(result); err != nil {
-		return fmt.Errorf("Error setting health watches: %s", err)
+		return fmt.Errorf("Error setting health watches: %w", err)
 	}
 	return
 }
 
+// HealthGet retrieve the current state of the DCGM health check system
+func HealthGet(groupId GroupHandle) (HealthSystem, error) {
+	var systems C.dcgmHealthSystems_t
+
+	result := C.dcgmHealthGet(handle.handle, groupId.handle, (*C.dcgmHealthSystems_t)(unsafe.Pointer(&systems)))
+	if err := errorString(result); err != nil {
+		return HealthSystem(0), err
+	}
+	return HealthSystem(systems), nil
+}
+
+type IncidentError struct {
+	Message string
+	Code    HealthCheckError
+}
+
+type Incident struct {
+	System     HealthSystem
+	Health     HealthResult
+	Error      IncidentError
+	EntityInfo GroupEntityPair
+}
+
+type HealthResponse struct {
+	OverallHealth HealthResult
+	Incidents     []Incident
+}
```

Comment:
I like how you are capturing the right details, however some details like `Code` and `EntityInfo` are not used today. Is it done to re-use the HealthResponse struct to provide more details in future?

Reply:
@rohit-arora-dev , The goal to expose the full response to go-dcgm package consumers, such as the dcgm-exporter.